projects/Mallard/src/hooks/apollo.ts

```diff
@@ -0,0 +1,23 @@
+import { useQuery as useApolloQuery } from '@apollo/react-hooks'
+import ApolloClient from 'apollo-client'
+import { DocumentNode } from 'graphql'
+
+/**
+ * Light wrapper around Apollo own `useQuery` to directly throw errors (to be
+ * caught by the closest exception handler). This allows callsites to have
+ * less boilerplate code and is more idiomatic modern React.
+ *
+ * This return `undefined` while the query is being resolved (ex. a fetch is in
+ * progress). Once all the data is available, the component re-renders and
+ * `useQuery` will return the final data.
```

Comment:
After speaking IRL with @jeanlauliac point 2 is handled nicely by Apollo 🚀

Reply:
> Do we always want to be communicating in errors here?

In general we shouldn't let any error fall through or be silent, we always want them to reach Sentry. I also believe it's better to crash early rather than having a UI getting in some corrupted state. To render errors just for some parts of the UI, we can use error boundaries. For example, we should probably have one around the weather module so that if it crashes for any reasons, only this crashes.  This is better because it handles errors from any sources, including errors within `render()` functions themselves.

Reply:
> How does this hook deal with the second load. It looks like it will return undefined rather than show the stale data, so you'll get a flash of no content between each load?

In that particular PR case we don't have second renders. But, for data that might be fetched from GraphQL for example, I can confirm Apollo would not go back in loading state. It keeps showing the stale data instead, and replace only when fetched. However, it does provide a flag so that we can have  **both**, say, a loading spinner, and the stale data still showing.